### PR TITLE
Issue 17671 cud for language resource

### PR DESCRIPTION
**Pull request by dsilvam**
New endpoints for the LanguageResource v2 were created to support:
1. Saving a new Language
2. Updating an existing Language
3. Deleting an existing Language

Also `DotRuntimeException` was changed to `DoesNotExistException` to have the ResponseUtil return the correct response code (404) instead of 500 when not finding a Language by id. 

Finally, postman tests were included for the three new endpoints for different scenarios, success and failures.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java [hunk 1]
```diff
@@ -73,6 +82,94 @@ public Response  list(@Context final HttpServletRequest request, @Context final
...
+    public final Response saveLanguage(@Context final HttpServletRequest request,
+            @Context final HttpServletResponse response,
+            final LanguageForm languageForm) {
+        this.webResource.init(null, request, response,
+                true, PortletID.LANGUAGES.toString());
+        try {
+            DotPreconditions.notNull(languageForm,"Expected Request body was empty.");
+            final Language language = saveOrUpdateLanguage(null, languageForm);
+            return Response.ok(new ResponseEntityView(language)).build(); // 200
+        } catch (Exception e) {
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java]: The exceptions should be handle by the Mappers

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java [hunk 2]
```diff
@@ -73,6 +82,94 @@ public Response  list(@Context final HttpServletRequest request, @Context final
...
+            @PathParam("languageId") final String languageId,
+            final LanguageForm languageForm) {
+        this.webResource.init(null, request, response,
+                true, PortletID.LANGUAGES.toString());
+        try {
+            DotPreconditions.checkArgument(UtilMethods.isSet(languageId),"Language Id is required.");
+            DotPreconditions.notNull(languageForm,"Expected Request body was empty.");
+            final Language language = saveOrUpdateLanguage(languageId, languageForm);
+            return Response.ok(new ResponseEntityView(language)).build(); // 200
+        } catch (Exception e) {
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java]: same before

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
+package com.dotcms.rest.api.v2.languages;
+
+import com.dotcms.repackage.javax.validation.constraints.NotNull;
+import com.dotcms.repackage.javax.validation.constraints.Size;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.repackage.javax.validation.constraints.Size'](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    public static final class Builder {
+
+        @JsonProperty
+        private String languageCode;
+
+        @JsonProperty
+        private String countryCode;
+
+        @JsonProperty
+        private String language;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Field language has the same name as a method](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+        @JsonProperty
+        private String country;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Field country has the same name as a method](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    public String getCountry() {
+        return country;
+    }
+
+    public static final class Builder {
+
+        @JsonProperty
+        private String languageCode;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Field languageCode has the same name as a method](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java [hunk 7]
```diff
@@ -73,6 +82,94 @@ public Response  list(@Context final HttpServletRequest request, @Context final
...
+     * Persists a new {@link Language}
+     *
+     * @param request HttpServletRequest
+     * @param languageForm LanguageForm
+     * @return Response
+     */
+    @POST
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java]: Codacy Issue found: [The String literal "application/javascript" appears 4 times in this file; the first occurrence is on line 95](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java [hunk 8]
```diff
@@ -73,6 +82,94 @@ public Response  list(@Context final HttpServletRequest request, @Context final
         return Response.ok(new ResponseEntityView(ImmutableList.copyOf(languages))).build();
     }
 
+    /**
+     * Persists a new {@link Language}
+     *
+     * @param request HttpServletRequest
+     * @param languageForm LanguageForm
+     * @return Response
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java]: how is the content as this Response?

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java [hunk 9]
```diff
@@ -0,0 +1,83 @@
+package com.dotcms.rest.api.v2.languages;
+
+import com.dotcms.repackage.javax.validation.constraints.NotNull;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.repackage.javax.validation.constraints.NotNull'](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+        return country;
+    }
+
+    public static final class Builder {
+
+        @JsonProperty
+        private String languageCode;
+
+        @JsonProperty
+        private String countryCode;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Field countryCode has the same name as a method](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java [hunk 11]
```diff
@@ -83,4 +180,26 @@ public Response getMessages(@Context HttpServletRequest request,
...
     }
+
+    private Language saveOrUpdateLanguage(final String languageId, final LanguageForm form) {
+        final Language newLanguage = new Language();
+
+        if (StringUtils.isSet(languageId)) {
+            try {
+                final Language origLanguage = this.languageAPI.getLanguage(languageId);
+                BeanUtils.copyProperties(newLanguage, origLanguage);
+            } catch (Exception e) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java]: Codacy Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+
+    private final String languageCode;
+
+    private final String countryCode;
+
+    private final String language;
+
+    private final String country;
+
+    public LanguageForm(final LanguageForm.Builder builder) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [It is a good practice to call super() in a constructor](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java [hunk 13]
```diff
@@ -73,6 +82,94 @@ public Response  list(@Context final HttpServletRequest request, @Context final
...
+        }
+    }
+
+    /**
+     * Updates an already persisted {@link Language}
+     *
+     * @param request HttpServletRequest
+     * @param languageId languageId
+     * @param languageForm LanguageForm
+     * @return Response
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java]: how is the content as this Response?

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java [hunk 14]
```diff
@@ -73,6 +82,94 @@ public Response  list(@Context final HttpServletRequest request, @Context final
...
+            @Context final HttpServletResponse response,
+            @PathParam("languageId") final String languageId) {
+        this.webResource.init(null, request, response,
+                true, PortletID.LANGUAGES.toString());
+        try {
+            DotPreconditions.checkArgument(UtilMethods.isSet(languageId),"Language Id is required.");
+            final Language language = languageAPI.getLanguage(languageId);
+            languageAPI.deleteLanguage(language);
+            return Response.ok(new ResponseEntityView(OK)).build(); // 200
+        } catch (Exception e) {
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java]: same before

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java [hunk 15]
```diff
@@ -73,6 +82,94 @@ public Response  list(@Context final HttpServletRequest request, @Context final
...
+            return ResponseUtil.mapExceptionResponse(e);
+        }
+    }
+
+    /**
+     * Deletes an already persisted {@link Language}
+     *
+     * @param request HttpServletRequest
+     * @param languageId languageId
+     * @return Response
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java]: same before

File: dotCMS/src/main/java/com/dotmarketing/portlets/languagesmanager/business/LanguageAPIImpl.java [hunk 16]
```diff
@@ -159,6 +160,11 @@ public Language getLanguage(final long id) {
 	@WrapInTransaction
 	@Override
 	public void saveLanguage(final Language language) {
+		DotPreconditions.checkNotNull(language, "Language can't be null");
+		DotPreconditions.checkArgument(UtilMethods.isSet(language.getLanguageCode()),
+				"Language Code can't be null or empty");
+		DotPreconditions.checkArgument(UtilMethods.isSet(language.getLanguage()),
+				"Language String can't be null or empty");
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/portlets/languagesmanager/business/LanguageAPIImpl.java]: test?

File: dotCMS/src/main/java/com/dotmarketing/portlets/languagesmanager/business/LanguageFactoryImpl.java [hunk 17]
```diff
@@ -134,7 +135,7 @@ protected Language getLanguage(final String languageId) {
 			return getLanguage(codes[0], codes[1]);
 		} catch (Exception e) {
 			Logger.error(LanguageFactoryImpl.class, "getLanguage failed for id:" + languageId,e);
-			throw new DotRuntimeException("getLanguage failed for id:" + languageId, e);
+			throw new DoesNotExistException("getLanguage failed for id:" + languageId, e);
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/portlets/languagesmanager/business/LanguageFactoryImpl.java]: test?

**Review by freddyucv — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java [hunk 18]
```diff
@@ -83,4 +180,26 @@ public Response getMessages(@Context HttpServletRequest request,
...
+    private Language saveOrUpdateLanguage(final String languageId, final LanguageForm form) {
+        final Language newLanguage = new Language();
+
+        if (StringUtils.isSet(languageId)) {
+            try {
+                final Language origLanguage = this.languageAPI.getLanguage(languageId);
+                BeanUtils.copyProperties(newLanguage, origLanguage);
+            } catch (Exception e) {
+                Logger.debug(this.getClass(), () -> "Unable to find language with id:" + languageId);
+                throw new DoesNotExistException(e.getMessage(), e);
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java]: Perhaps, the exception name is not the most accurate

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java [hunk 19]
```diff
@@ -1,11 +1,22 @@
 package com.dotcms.rest.api.v2.languages;
 
+import static com.dotcms.rest.ResponseEntityView.OK;
+
 import com.dotcms.repackage.com.google.common.annotations.VisibleForTesting;
+import com.dotcms.rest.api.v1.authentication.ResponseUtil;
+import com.dotcms.util.DotPreconditions;
+import com.dotmarketing.exception.DoesNotExistException;
+import com.dotmarketing.exception.DotRuntimeException;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.exception.DotRuntimeException'](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java (hunk 9 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.repackage.javax.validation.constraints.NotNull'](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java (hunk 10 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Field countryCode has the same name as a method](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.repackage.javax.validation.constraints.Size'](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Field language has the same name as a method](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Field languageCode has the same name as a method](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java [hunk 20]
```diff
@@ -73,6 +84,75 @@ public Response  list(@Context final HttpServletRequest request, @Context final
...
+     * Persists a new {@link Language}
+     *
+     * @param request HttpServletRequest
+     * @param languageForm LanguageForm
+     * @return Response
+     */
+    @POST
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java]: Codacy Issue found: [The String literal "application/javascript" appears 4 times in this file; the first occurrence is on line 95](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java [hunk 21]
```diff
@@ -1,11 +1,22 @@
 package com.dotcms.rest.api.v2.languages;
 
+import static com.dotcms.rest.ResponseEntityView.OK;
+
 import com.dotcms.repackage.com.google.common.annotations.VisibleForTesting;
+import com.dotcms.rest.api.v1.authentication.ResponseUtil;
+import com.dotcms.util.DotPreconditions;
+import com.dotmarketing.exception.DoesNotExistException;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.exception.DoesNotExistException'](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java [hunk 22]
```diff
@@ -1,11 +1,22 @@
 package com.dotcms.rest.api.v2.languages;
 
+import static com.dotcms.rest.ResponseEntityView.OK;
+
 import com.dotcms.repackage.com.google.common.annotations.VisibleForTesting;
+import com.dotcms.rest.api.v1.authentication.ResponseUtil;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.rest.api.v1.authentication.ResponseUtil'](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java (hunk 12 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [It is a good practice to call super() in a constructor](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Field country has the same name as a method](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java [hunk 23]
```diff
@@ -83,4 +165,26 @@ public Response getMessages(@Context HttpServletRequest request,
...
+        newLanguage.setLanguageCode(form.getLanguageCode());
+        newLanguage.setLanguage(form.getLanguage());
+        newLanguage.setCountryCode(form.getCountryCode());
+        newLanguage.setCountry(form.getCountry());
+
+        this.languageAPI.saveLanguage(newLanguage);
+        return newLanguage;
+    }
+
+    private boolean doesLanguageExist(String languageId) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java]: Codacy Issue found: [Parameter 'languageId' is not assigned and could be declared final](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Field country has the same name as a method](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.repackage.javax.validation.constraints.Size'](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java (hunk 10 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Field countryCode has the same name as a method](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java (hunk 9 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.repackage.javax.validation.constraints.NotNull'](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java [hunk 24]
```diff
@@ -73,6 +84,77 @@ public Response  list(@Context final HttpServletRequest request, @Context final
...
+     * Persists a new {@link Language}
+     *
+     * @param request HttpServletRequest
+     * @param languageForm LanguageForm
+     * @return Response
+     */
+    @POST
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java]: Codacy Issue found: [The String literal "application/javascript" appears 4 times in this file; the first occurrence is on line 95](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Field languageCode has the same name as a method](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/languagesmanager/business/LanguageFactoryIntegrationTest.java [hunk 25]
```diff
@@ -12,6 +12,7 @@
 import com.dotcms.repackage.org.apache.struts.Globals;
 import com.dotcms.util.IntegrationTestInitService;
 import com.dotmarketing.business.FactoryLocator;
+import com.dotmarketing.exception.DoesNotExistException;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/languagesmanager/business/LanguageFactoryIntegrationTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.exception.DoesNotExistException'](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java (hunk 19 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.exception.DotRuntimeException'](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java (hunk 12 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [It is a good practice to call super() in a constructor](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Field language has the same name as a method](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java (hunk 22 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.rest.api.v1.authentication.ResponseUtil'](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotmarketing/portlets/languagesmanager/business/LanguageAPIImpl.java (hunk 16 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotmarketing/portlets/languagesmanager/business/LanguageAPIImpl.java]: Done

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java [hunk 26]
```diff
@@ -73,6 +88,91 @@ public Response  list(@Context final HttpServletRequest request, @Context final
...
+     *  <li>{@code languageCode}
+     *  <li>{@code countryCode}
+     *  <li>{@code language}
+     *  <li>{@code country}
+     *  </ul>
+     */
+    @POST
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java]: Codacy Issue found: [The String literal "application/javascript" appears 4 times in this file; the first occurrence is on line 95](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java [hunk 27]
```diff
@@ -0,0 +1,81 @@
...
+    public static final class Builder {
+
+        @JsonProperty
+        private String languageCode;
+
+        @JsonProperty
+        private String countryCode;
+
+        @JsonProperty
+        private String language;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Field language has the same name as a method](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java [hunk 28]
```diff
@@ -0,0 +1,81 @@
...
+        return country;
+    }
+
+    public static final class Builder {
+
+        @JsonProperty
+        private String languageCode;
+
+        @JsonProperty
+        private String countryCode;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Field countryCode has the same name as a method](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java [hunk 29]
```diff
@@ -0,0 +1,81 @@
...
+
+        @JsonProperty
+        private String language;
+
+        @JsonProperty
+        private String country;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Field country has the same name as a method](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java [hunk 30]
```diff
@@ -0,0 +1,81 @@
...
+
+    private final String languageCode;
+
+    private final String countryCode;
+
+    private final String language;
+
+    private final String country;
+
+    public LanguageForm(final LanguageForm.Builder builder) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [It is a good practice to call super() in a constructor](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java [hunk 31]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    public String getCountry() {
+        return country;
+    }
+
+    public static final class Builder {
+
+        @JsonProperty
+        private String languageCode;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Field languageCode has the same name as a method](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java (hunk 31 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Field languageCode has the same name as a method](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java (hunk 30 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [It is a good practice to call super() in a constructor](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java (hunk 28 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Field countryCode has the same name as a method](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java (hunk 29 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Field country has the same name as a method](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java (hunk 27 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguageForm.java]: Codacy Issue found: [Field language has the same name as a method](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java (hunk 26 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java]: Codacy Issue found: [The String literal "application/javascript" appears 4 times in this file; the first occurrence is on line 95](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)

**Pull request merged**